examples/platform/esp32/time/TimeSync.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TimeSync.h"
+#include <esp_sntp.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#define REF_TIME 1546300800 /* 01-Jan-2019 00:00:00 */
+/* Timer interval once every day (24 Hours) */
+#define TIME_PERIOD_SEC 86400000ULL
+namespace {
+const uint8_t kMaxNtpServerStringSize = 128;
+char mSntpServerName[kMaxNtpServerStringSize];
+
+static CHIP_ERROR GetLocalTimeString(char * buf, size_t buf_len)
+{
+    struct tm timeinfo;
+    char strftime_buf[64];
+    time_t now;
+    time(&now);
+    localtime_r(&now, &timeinfo);
+    strftime(strftime_buf, sizeof(strftime_buf), "%c %z[%Z]", &timeinfo);
```

Comment:
Do not use `%c` for formatting time. Use the ISO8601 format:

`%Y-%m-%dT%H:%M:%S%z`

Also, what is the current timezone in your implementation? How are they set? I am concerned that using local time here rather than UTC time may lead to incorrect times.

Reply:
@tcarmelveilleux  I have not set any timezone here. Just fetching time from the server.

Reply:
Even if I don’t set the timezone, my device is having UTC time.